src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -378,10 +378,16 @@ internal static bool IsCompilerServerSupported(string tempPath)
             }
         }
 
-        internal static bool TryCreateServerCore(string clientDir, string pipeName)
+        internal static (string processFilePath, string commandLineArguments, string toolFilePath) GetServerProcessInfo(string clientDir, string pipeName)
         {
             var serverPathWithoutExtension = Path.Combine(clientDir, "VBCSCompiler");
-            var serverInfo = RuntimeHostInfo.GetProcessInfo(serverPathWithoutExtension, $"-pipename:{pipeName}");
+            var commandLineArgs = $@"""-pipename:{pipeName}""";
```

Comment:
Should the quotes wrap the name only? `-pipename:""{pipename}""`

Reply:
Yes / No. Both quoting the entire argument, or quoting the actual name, will work here. Quoted the entire argument as this was the previous behavior + it's easier to verify the functionality cross platform (walked through the process argument parsing code).